### DISCUSSION TITLE
How to check localStorage property and make api call based on localStorage property

### DISCUSSION THREAD

**Discussion by deKuddus** [vercel/next.js#32435] [Help]
Hi trying to upgrade to next 12 and also want to use middleware features in our project. So what I need to do , I have to check a localStorage token property. If the token exist then we will check the token is expired or not. If token is expired the user will be logout and have to re login. If token is valid than they can go next request.

so when I write `localStorage.get('token') ` it says `localStorage` not defined.  Does any one here to help me out please???

**Comment by rafde** [comment]
@deKuddus `localStorage` not defined usually happens when it's trying to use it in server side.
The code can work if you do a check for server-side. Example

```jsx
React.useEffect( () => {
  if ( !globalThis.localStorage ) {
    return;
  }

  const token = localStorage.get('token');
  // do server side stuff
}, []);
```
